seed/internal/validate.go

```diff
@@ -0,0 +1,36 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package internal
+
+import (
+	"fmt"
+	"regexp"
+)
+
+var validSeedSystemLabel = regexp.MustCompile("^[a-zA-Z0-9](?:-?[a-zA-Z0-9])+$")
```

Comment:
This regex has apparently been tweaked slightly, not just moved, and it now requires 2 characters where the old one would accept a single one if I'm reading this right. I suppose that's perfectly fine, but mentioning just in case.

Reply:
Yup. One letter label is IMO too short.

Reply:
the regexp was intentionally copying 

var validModel = regexp.MustCompile("^[a-zA-Z0-9](?:-?[a-zA-Z0-9])*$")

from asserts/model.go

While our conventions and tooling will tend to use date/time based labels, model named ones might also make sense.

Reply:
Pushe the tweak